### PR TITLE
Introduce a compatibility layer for integration specific code

### PR DESCRIPTION
**Pull request by kasparsd**
Builds on #308.

### CODE REVIEW

**Comment by kasparsd** [comment]
@pyronaur What are your thoughts on this implementation?

File: class-two-factor-compat.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	 *
+	 * @return void
+	 */
+	public function init() {
+		/**
+		 * Jetpack
+		 *
+		 * @see https://wordpress.org/plugins/jetpack/
+		 */
+		add_filter( 'two_factor_rememberme', array( $this, 'jetpack_rememberme' ) );
```
Comment (pyronaur) [inline class-two-factor-compat.php]: Maybe add a condition to only attach the hook if Jetpack is installed?

**Comment by kasparsd** [comment]
Thanks for the review and feedback @pyronaur! Were you also able to test it in your dev environment?

File: class-two-factor-compat.php (hunk 1 above)
Comment (kasparsd) [reply class-two-factor-compat.php]: Were you thinking of checking for the presence of the `Jetpack` class or one of the constants `JETPACK__VERSION` or `JETPACK__PLUGIN_FILE`? The `is_plugin_active()` check could fail if Jetpack is installed in a different directory, for example (during dev or some custom plugin tree).

Comment (pyronaur) [reply class-two-factor-compat.php]: I was thinking testing `Jetpack` class presence

**Comment by pyronaur** [comment]
Give me a few moments - I'll give it a few tests.

**Comment by pyronaur** [comment]
@kasparsd Tested this just now - it works!

**Pull request merged**